### ISSUE TITLE
hpack warnings are not errors with --pedantic

### ISSUE THREAD

**Issue by quasicomputational** [commercialhaskell/stack#3772]
hpack can generate warnings (e.g., for unrecognised fields). It'd be in the spirit of `--pedantic` to fail for those as well, but currently they're just printed and the build carries on.

**Comment by mgsloan** [comment]
Yeah extending the meaning of --pedantic probably makes sense.  Note that right now the only warnings it turns into errors are GHC's.
